crates/wasi-preview1-component-adapter/Cargo.toml

```diff
@@ -0,0 +1,58 @@
+[package]
+name = "wasi-preview1-component-adapter"
+version.workspace = true
+authors.workspace = true
+edition.workspace = true
+
+[dependencies]
+wasi = { version = "0.11.0", default-features = false }
+wit-bindgen = { version = "0.4.0", default-features = false, features = ["macros"] }
+byte-array-literals = { workspace = true }
+
+[build-dependencies]
+wasm-encoder = "0.25"
+object = { version = "0.30.0", default-features = false, features = ["archive"] }
+
+[lib]
+crate-type = ["cdylib"]
+
+[profile.release]
+# Omit any unwinding support. This is currently the default for wasm, though
+# that could theoretically change in the future.
+panic = 'abort'
+opt-level = 's'
+strip = 'debuginfo'
+
+# Make dev look like a release build since this adapter module won't work with
+# a debug build that uses data segments and such.
+[profile.dev]
+panic = 'abort'
+incremental = false
+opt-level = 's'
+
+# Omit assertions, which include failure messages which require string
+# initializers.
+debug-assertions = false
+
+# Omit integer overflow checks, which include failure messages which require
+# string initializers.
+overflow-checks = false
+
+[features]
+default = ["reactor"]
+reactor = []
+command = []
+
+[workspace]
```

Comment:
Could this get removed so this is integrated into the main wasmtime workspace? I think the above fiddly bits about profiles and such can go into the main workspace as well and apply to just this one crate.

Reply:
I would love to put it all in one workspace, yeah. I wasn't able to figure out how to apply these profiles to just one crate, but if we can do it then

Reply:
I think [profile overrides](https://doc.rust-lang.org/nightly/cargo/reference/profiles.html#overrides) show enable that?

Reply:
That didn't work because
> Overrides cannot specify the `panic` ... settings
but I just realized, since that we have to avoid the panic machinery for vtable reasons, maybe the adapter will also build correctly without that in the profile...

Reply:
oh that should also be ok because wasm doesn't actually support panic=unwind vs panic=abort, it's just always panic=abort anyway. So that setting can be dropped without affecting the output.

Reply:
Thanks!